### ISSUE TITLE
Requestion: Tensorboard 1d graphs update view along with "Reload Data" setting or newly added setting option.

### ISSUE THREAD

**Issue by isaacgerg** [tensorflow/tensorboard#6668]
Currently, tensoboard graphs update automated when "Reload Data" is checked in the settings.  However, the graph will continue "to the right" of the viewport.  This makes the latest data occluded from the user, which is likely what the user wants to see most.  Propose to improve upon this behavior by invoking the viewport reset that is done when a user "double clicks" the viewport, and make this an additional setting.  The setting can be off by default to not interfere with existing workflows that expect the current behavior.  For others, selecting this new option will reset the view port in the same manner as double clicking the graph every time a refresh of the data is done for display.

**Comment by bmd3k** [comment]
Hi, if I understand the request correctly, I have just confirmed that TensorBoard should behave the way you want. I tested with tensorboard 2.14.1 and 2.15.0. I tested both the "Time Series" tab and the "Scalars" tab.

So, for example:
1. I load the following in Firefox (it is currently broken in Chrome):
  * https://colab.sandbox.google.com/github/tensorflow/tensorboard/blob/master/docs/get_started.ipynb
 
2. I run the cells that `%load_ext tensorboard` and `%tensorboard --logdir logs/fit` before running any of the other cells. This loads a tensorboard without any plots.

3. I configure the TensorBoard to reload every 30 seconds.

4. I run the cells that train the model.

5. I observe that, over time, the TensorBoard updates with plots and the x-axis of the plots expand to cover all the steps/epochs as training continues.

![image](https://github.com/tensorflow/tensorboard/assets/17152369/ff088574-e8d0-420e-a8d8-15e89264191b)

**Comment by isaacgerg** [comment]
Your first step indicates the problem with my setup; I am using chrome.  I verified that it is broken in chrome but not in firefox.  Since chrome was a google product like tensorboard, I assumed that tensorboard would work correctly and hence did not check with another browser.  Thank you and you may close.

**Comment by bmd3k** [comment]
Just to clarify. When I say "it is broken in Chrome" - i mean, tensorboard integration with colab is generally broken (See: [googlecolab/colabtools#3990](https://github.com/googlecolab/colabtools/issues/3990)). I expect that the specific functionality you are requesting should work in Chrome but haven't tested it out specifically.

**Comment by isaacgerg** [comment]
Understood.  It looks as if the autoupdate doesn't work in chome but works in firefox.  I am using tensboard 2.15.0.  Chrome Version 118.0.5993.118 (Official Build) (64-bit). Firefox version 118.0.1 (64-bit).

**Comment by isaacgerg** [comment]
Following up.  I let a torch session train over the weekend and upon returning, Firefox 118.0.1 (64-bit) did not update the x-axis on my 2 displays despite "Reload data" being checked.  Both graphs had this viewport setting selected: 
![image](https://github.com/tensorflow/tensorboard/assets/11971499/86fb98e3-543e-4529-8f9b-770d0cf689d4)

**Comment by qihach64** [comment]
Hi @isaacgerg thanks for the follow up! I tried to reproduce the issue following [Brain's previous test procedures](https://github.com/tensorflow/tensorboard/issues/6668#issuecomment-1789649345) and things worked just fine for Chrome as well now that the integration [bug](https://github.com/googlecolab/colabtools/issues/3990) with Colab is fixed. 

I hope to get clarification on a few things to help troubleshooting this:
1. Could you add a screenshot or video recording of the issue you ran into? We want to make sure we can reproduce it first.
2. Does the reloading work fine on TimeSeries tab and only stops working on Scalars tab (here is a [snapshot with TimeSeries and Scalar tab](https://screenshot.googleplex.com/BDPwCJdbvUVq2Ah)) ? Or do both tabs have the same behavior?
3. Could you reproduce the issue when training the simple MNIST data set from the [Get Started](https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/get_started.ipynb) example that Brian linked? This is to verify that the issue is an isolated FE issue rather than something caused by the particular data set.

Thank you for your time and patience!

**Comment by isaacgerg** [comment]
1. Below is a screenshot.  I am using tensorboard 2.15.0 and firefox 118.0.1 on Windows 10.  See that for val_loss the graph continues along the x-axis off the viewport.  You can see from the above graph (which I double clicked on to reset the view) that there are more iterations than visible in the viewport (both metrics have the same number of items).

![image](https://github.com/tensorflow/tensorboard/assets/11971499/ae5beb8d-e68d-4ab2-8cd3-bfff1415892c)

2. The bug does not occur with the time series option, only the scalars option.
3. Unfortunately I don't have the time to do thisl.

**Comment by bmd3k** [comment]
The Scalars dashboard is in maintenance mode (for several years now) and our efforts are instead focused on the Time Series dashboard. My recommendation would be to use Time Series dashboard.

**Comment by isaacgerg** [comment]
Might be helpful to put a note on the scalars board that its in matinananced mode.  I have used tensoboard board since it was first released and never knew about the state of the scalars board.  I continue to use it because its simply and no fluff which is what i need for my work.

**Comment by qihach64** [comment]
> Might be helpful to put a note on the scalars board that its in matinananced mode. I have used tensoboard board since it was first released and never knew about the state of the scalars board. I continue to use it because its simply and no fluff which is what i need for my work.

This is a good suggestion, but we are not quite ready to claim that the Scalars dashboard is in maintenance mode yet. I will leave this issue open in case more people are running into this and we can prioritize on fixing this for Scalars dashboard then.

**Comment by qihach64** [comment]
> Might be helpful to put a note on the scalars board that its in matinananced mode

That's a good suggestion and we will consider adding that for further clarification. 

I will close this issue for now.
